spec/rspec/core/memoized_helpers_spec.rb

```diff
@@ -391,6 +391,44 @@ def count
       expect(counter.count).to eq(2)
     end
 
+    it "is threadsafe" do
+      value_queue = Queue.new
+      values      = []
+      RSpec.describe do
+        # a let block that will block
+        let(:memoize_threadsafe) { value_queue.shift }
+
+        example do
+          thread_blocked = Queue.new
+
+          # blocked, waiting on let to return
+          thread1 = Thread.new do
+            thread_blocked << true
+            values << memoize_threadsafe
+          end
+          thread_blocked.shift
+
+          # blocked, waiting on let to return
+          thread2 = Thread.new do
+            thread_blocked << true
+            values << memoize_threadsafe
+          end
+          thread_blocked.shift
+
+          # provide the values they are blocking for
+          value_queue << :value1
+          value_queue << :value2
```

Comment:
WDYT about using symbols that describe the roles these two play rather than the more generic `:value1` and `:value2`?  Something like:

``` ruby
value_queue << :value_let_should_return
value_queue << :value_let_should_not_return_if_threadsafe
```